homeassistant/components/sensor/hydroquebec.py

```diff
@@ -0,0 +1,297 @@
+
+SENSOR_TYPES = {
+    'period_total_bill': ['Current period bill',
+                          PRICE, 'mdi:square-inc-cash'],
+    'period_length': ['Current period length',
+                      DAYS, 'mdi:calendar-today'],
+    'period_total_days': ['Total number of days in this period',
+                          DAYS, 'mdi:calendar-today'],
+    'period_mean_daily_bill': ['Period daily average bill',
+                               PRICE, 'mdi:square-inc-cash'],
+    'period_mean_daily_consumption': ['Period daily average consumption',
+                                      KILOWATT_HOUR, 'mdi:flash'],
+    'period_total_consumption': ['Total Consumption',
+                                 KILOWATT_HOUR, 'mdi:flash'],
+    'period_lower_price_consumption': ['Period Lower price consumption',
+                                       KILOWATT_HOUR, 'mdi:flash'],
+    'period_higher_price_consumption': ['Period Higher price consumption',
+                                        KILOWATT_HOUR, 'mdi:flash'],
+    'yesterday_total_consumption': ['Yesterday total consumption',
+                                    KILOWATT_HOUR, 'mdi:flash'],
+    'yesterday_lower_price_consumption': ['Yesterday lower price consumption',
+                                          KILOWATT_HOUR, 'mdi:flash'],
+    'yesterday_higher_price_consumption':
+    ['Yesterday higher price consumption', KILOWATT_HOUR, 'mdi:flash'],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_MONITORED_VARIABLES):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+HOST = "https://www.hydroquebec.com"
+HOME_URL = HOST + "/portail/web/clientele/authentification"
+PROFILE_URL = HOST + "/portail/fr/group/clientele/portrait-de-consommation"
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Bbox sensor."""
+    # Create a data fetcher to support all of the configured sensors. Then make
+    # the first call to init the data.
+
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    try:
+        hydroquebec_data = HydroquebecData(username, password)
+        hydroquebec_data.update()
+    except requests.exceptions.HTTPError as error:
+        _LOGGER.error(error)
+        return False
+
+    name = config.get(CONF_NAME)
+
+    sensors = []
+    for variable in config[CONF_MONITORED_VARIABLES]:
+        sensors.append(HydroQuebecSensor(hydroquebec_data, variable, name))
+
+    add_devices(sensors)
+
+
+class HydroQuebecSensor(Entity):
+    """Implementation of a HydroQuebec sensor."""
+
+    def __init__(self, hydroquebec_data, sensor_type, name):
+        """Initialize the sensor."""
+        self.client_name = name
+        self.type = sensor_type
+        self.entity_id = "sensor." + name + "_" + sensor_type
```

Comment:
I'm overriding the entity_id because of really long description of some sensors (like `Total number of days in this period`)
This is not a good usage of entity_id attribute ?